### PR TITLE
fix unstable mpp fail tests

### LINKED ISSUE #5708: mpp_fail.test is unstable
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
`mpp_fail.test` is unstable
```
[2022-08-25T09:14:49.285Z]   File: fullstack-test/mpp/mpp_fail.test

[2022-08-25T09:14:49.285Z]   Error line: 79

[2022-08-25T09:14:49.285Z]   Error: use test; set @@tidb_isolation_read_engines='tiflash'; set @@tidb_allow_mpp=1; select count(value), id from t group by id;

[2022-08-25T09:14:49.285Z]   Result:

[2022-08-25T09:14:49.285Z]     ERROR 1105 (HY000) at line 1: other error for mpp stream: From MPP<query:435530993011261446,task:1>: Code: 0, e.displayText() = DB::Exception: Consumer exits unexpected, Receiver closed, e.what() = DB::Exception,

[2022-08-25T09:14:49.285Z]   Expected:

[2022-08-25T09:14:49.285Z]     {#REGEXP}.*Fail point FailPoints::exception_during_mpp_root_task_run is triggered.*
```

### PR DESCRIPTION
**Pull request by windtalker**
Signed-off-by: xufei <xufeixw@mail.ustc.edu.cn>

### What problem does this PR solve?

Issue Number: close #5708, ref #5095

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Mpp/MPPTask.cpp [hunk 1]
```diff
@@ -412,7 +412,11 @@ void MPPTask::runImpl()
         while (from->read())
             continue;
 
+        // finish DataStream
         from->readSuffix();
+        // finish receiver
+        receiver_set->close();
```
Comment (bestwoody) [inline dbms/src/Flash/Mpp/MPPTask.cpp:418]: what if a error throwed before

Comment (windtalker) [reply dbms/src/Flash/Mpp/MPPTask.cpp:418]: Then the error handle flow will abort the tunnels.

**Review by bestwoody — APPROVED** [review]
lgtm

**Review by SeaRise — APPROVED** [review]

**Pull request merged**